apm-agent-plugins/apm-ratpack-plugin/pom.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>apm-agent-plugins</artifactId>
+        <groupId>co.elastic.apm</groupId>
+        <version>1.7.1-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>apm-ratpack-plugin</artifactId>
+    <name>${project.groupId}:${project.artifactId}</name>
+
+    <properties>
+        <!-- Ratpack requires at least Java 8 -->
+        <maven.compiler.target>8</maven.compiler.target>
```

Comment:
My attempt to compile the Ratpack plugin at Java 8. Need guidance as to appropriate approach.

Reply:
The compiler target should still be 7. Ratpack will still be capable of running Java 7 bytecode on a Java 8 JVM. As long as you are not using Java 8 bytecode instructions (for example lambdas) you can still compile it with bytecode level targeting Java 7. Obviously, you'll need at least a JDK 8 to compile which has, for example, `java.util.function.Consumer` in the classpath. But we already require that in other places.

Reply:
OK. I take what you're saying to mean that I should be able to use `java.util.Optional` then, but as long as I use anonymous classes in place of lambdas? However, the animal-sniffer plugin is complaining when I do. Is there a way to resolve this?

[INFO] --- animal-sniffer-maven-plugin:1.17:check (signature-check) @ apm-ratpack-plugin ---
[INFO] Checking unresolved references to org.codehaus.mojo.signature:java17:1.0
[ERROR] Undefined reference: java.util.Optional
[ERROR] Undefined reference: void java.util.Optional.ifPresent(java.util.function.Consumer)
[ERROR] Undefined reference: boolean java.util.Optional.isPresent()

Reply:
I ran across the `import org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement` annotation, which appears to have done the trick (I'm assuming that's the correct approach). Maven build appears to be working.